src/main/java/io/confluent/idesidecar/restapi/clients/SchemaErrors.java

```diff
@@ -9,39 +9,58 @@
 import java.time.Duration;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * The `SchemaErrors` class manages schema error caching for different connections.
+ * It uses Caffeine cache to store and retrieve schema errors based on connection and schema IDs.
+ *
+ * <p>This class is application-scoped and observes connection lifecycle events to manage the cache.
+ * It provides methods to read, write, and clear schema errors for specific connections.
+ */
 
```

Comment:
nit: The Javadoc and the class definition should not be separated by whitespace.

Reply:
+1